internal/provider/resources/work_pool.go

```diff
@@ -42,7 +44,9 @@ type WorkPoolResourceModel struct {
 	BaseJobTemplate  types.String `tfsdk:"base_job_template"`
 }
 
-//nolint:ireturn
+// New returns a new WorkPoolResource.
```

Comment:
```suggestion
// NewWorkPoolResource returns a new WorkPoolResource.
```

Reply:
I missed this comment, will update in my next PR